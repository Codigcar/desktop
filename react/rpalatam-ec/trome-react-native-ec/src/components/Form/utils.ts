import { useCallback, useState } from 'react'

import { FormErrorsProps, FormErrorsReturn, ValidationErros } from './types'

export const useFormErrors = ({
  ref: formRef,
  schema,
}: FormErrorsProps): FormErrorsReturn => {
  const [validationErrors, setValidationErros] = useState<ValidationErros>({})

  const handleChangeField = useCallback(
    async (name, value) => {
      const error = formRef.current?.getFieldError(name)
      const hasError = error && error.length > 0
      try {
        await schema.validate({ [name]: value })
        if (hasError) {
          formRef.current?.setFieldError(name, '')
          setValidationErros((prev) => ({ ...prev, [name]: '' }))
        }
      } catch (err) {
        if (err.inner[name] && (!error || error !== err.inner[name])) {
          formRef.current?.setFieldError(name, err.inner[name])
          setValidationErros((prev) => ({ ...prev, [name]: err.inner[name] }))
        } else if (!err.inner[name] && hasError) {
          formRef.current?.setFieldError(name, '')
          setValidationErros((prev) => ({ ...prev, [name]: '' }))
        }
      }
    },
    [formRef, schema],
  )

  const setErrors = useCallback(
    (errors) => {
      formRef.current?.setErrors(errors)
      setValidationErros(errors)
    },
    [formRef],
  )

  return {
    errors: validationErrors,
    setFieldError: handleChangeField,
    setErrors,
  }
}
