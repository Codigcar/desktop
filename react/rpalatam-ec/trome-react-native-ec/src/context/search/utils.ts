import { FilterKeys } from './search'
import { Story } from '../../entities/Story'

export type SearchState = {
  filters: Map<FilterKeys, string>
  hasMore: boolean
  hasError: boolean
  page: number
  status: 'idle' | 'started' | 'done'
  results: Story[]
  orderInverse: boolean
}

export const initialState: SearchState = {
  filters: new Map(),
  hasMore: true,
  hasError: false,
  page: 0,
  status: 'idle',
  results: [],
  orderInverse: false,
}

type SearchAction =
  | { type: 'init' }
  | { type: 'failure' }
  | { type: 'reset' }
  | { type: 'restore' }
  | { type: 'clear'; payload?: { searchTerm?: string } }
  | {
      type: 'success'
      payload: {
        results: Story[]
        filters: Map<FilterKeys, string>
        orderInverse: boolean
        hasMore: boolean
        page: number
      }
    }

export const searchReducer = (
  state = initialState,
  action: SearchAction,
): SearchState => {
  switch (action.type) {
    case 'init':
      return { ...state, hasError: false, status: 'started' }
    case 'success': {
      const {
        payload: { results, filters, orderInverse, hasMore, page },
      } = action
      return {
        ...state,
        page,
        orderInverse,
        results,
        filters,
        hasMore,
        status: 'done',
        hasError: false,
      }
    }
    case 'clear': {
      const { payload } = action
      const nextFilters = new Map<FilterKeys, string>()
      if (payload?.searchTerm !== undefined) {
        nextFilters.set('term', payload?.searchTerm)
      }
      return { ...initialState, filters: nextFilters }
    }
    case 'failure':
      return { ...state, hasError: true, status: 'done' }
    case 'reset': {
      const hasResults = state.results.length > 0
      return { ...state, status: hasResults ? 'done' : 'idle' }
    }
    case 'restore': {
      return initialState
    }
    default:
      return state
  }
}
