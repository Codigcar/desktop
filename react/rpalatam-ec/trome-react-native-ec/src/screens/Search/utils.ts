export const computeEstimatedDate = (date: number): string => {
  const currentDate = new Date()
  const approx = currentDate.getDate() + date
  currentDate.setDate(approx)

  const month = `0${currentDate.getUTCMonth() + 1}`.slice(-2)
  const day = `0${currentDate.getUTCDate()}`.slice(-2)
  const year = currentDate.getUTCFullYear()

  return `${year}-${month}-${day}`
}

export const parseDate = (date?: Date): string | null => {
  if (!date) return null
  const month = `0${date.getUTCMonth() + 1}`.slice(-2)
  const day = `0${date.getUTCDate()}`.slice(-2)
  const year = date.getUTCFullYear()

  return `${year}-${month}-${day}`
}

export const timeFrames = [
  {
    value: '',
    label: 'Cualquier fecha',
  },
  {
    value: '-1',
    label: 'Últimas 24 horas',
  },
  {
    value: '-7',
    label: 'Última semana',
  },
  {
    value: '-30',
    label: 'Último mes',
  },
  {
    value: 'custom',
    label: 'Personalizado',
  },
]
