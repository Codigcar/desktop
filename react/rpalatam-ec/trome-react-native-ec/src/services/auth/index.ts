import PianoId from './PianoId'
import type { Authentication } from './auth.types'

const auth: Authentication = new PianoId()

export default auth
