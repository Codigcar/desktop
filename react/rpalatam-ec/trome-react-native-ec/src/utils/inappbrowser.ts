import { Linking } from 'react-native'
import InAppBrowserClassMethods, {
  InAppBrowserOptions,
} from 'react-native-inappbrowser-reborn'

export async function openInBrowser(
  url: string,
  options?: InAppBrowserOptions,
): Promise<void> {
  try {
    if (!/^https?:\/\//.test(url)) throw new Error('No http')
    const isAvailable = await InAppBrowserClassMethods.isAvailable()
    if (!isAvailable) throw new Error('No available')
    InAppBrowserClassMethods.open(url, options)
  } catch (err) {
    Linking.openURL(url)
  }
}
