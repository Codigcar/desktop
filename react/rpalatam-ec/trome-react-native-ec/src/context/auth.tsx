import analytics from '@react-native-firebase/analytics'
import crashlytics from '@react-native-firebase/crashlytics'
import { EventEmitter } from 'events'
import React, {
  createContext,
  useCallback,
  useContext,
  useEffect,
  useState,
} from 'react'
import { Linking } from 'react-native'

import { User } from '../entities/User'
import auth from '../services/auth'
import subscription from '../services/subscription'
import { googleSignOut, init, tokenExchange } from '../utils/auth'
import { ENABLE_AUTHENTICATION } from '../utils/flags'
import { sendPostMessageToWebview } from '../utils/refs'

type AuthListener = (isAuthenticated: boolean) => void

interface Context {
  isAuthenticated: boolean | undefined
  isSubscribed: boolean | undefined
  signin: (provider: string) => void
  signout: () => Promise<void>
  updateUserProfile: (userProfile: User) => Promise<void>
  onAuthenticationChange: (listener: AuthListener) => () => void
  user: User
  verifySubscription: () => Promise<void>
}

const AuthContext = createContext<Context>({} as Context)
const authEmitter = new EventEmitter()
const AUTH_EVENT_NAME = 'auth.event'

export const AuthProvider: React.FC = ({ children }) => {
  const [isAuthenticated, setIsAuthenticated] = useState<boolean>()
  const [user, setUser] = useState<User>({})
  const [subscriptor, setSubscriptor] = useState<boolean>()

  const onAuthenticationChange = useCallback((listener: AuthListener) => {
    authEmitter.on(AUTH_EVENT_NAME, listener)
    return function () {
      authEmitter.off(AUTH_EVENT_NAME, listener)
    }
  }, [])

  useEffect(() => {
    init()
    const restoreSession = async () => {
      try {
        if (!ENABLE_AUTHENTICATION) return setIsAuthenticated(false)
        const token = await auth.getToken()
        if (!token) return setIsAuthenticated(false)
        setIsAuthenticated(true)
      } catch (error) {
        if (error instanceof Error) crashlytics().recordError(error)
        setIsAuthenticated(false)
      }
    }
    restoreSession()
  }, [])

  const signin = useCallback((_: string) => {
    setIsAuthenticated(true)
    authEmitter.emit(AUTH_EVENT_NAME, true)
  }, [])

  const signout = useCallback(async () => {
    await auth.logout()
    await googleSignOut()
    await analytics().logEvent('logout')
    setIsAuthenticated(false)
    sendPostMessageToWebview('Home', { type: 'auth.LOGOUT' })
    authEmitter.emit(AUTH_EVENT_NAME, false)
  }, [])

  const updateUserProfile = useCallback(async (data: User) => {
    sendPostMessageToWebview('Home', {
      type: 'auth.PROFILE',
      payload: { ...data },
    })
    setUser(data)
  }, [])

  useEffect(() => {
    if (typeof isAuthenticated === 'undefined') return
    if (isAuthenticated) {
      auth
        .getUserProfile()
        .then((profile) => {
          sendPostMessageToWebview('Home', {
            type: 'auth.PROFILE',
            payload: { ...profile },
          })
          setUser(profile)
        })
        .catch((error) => {
          signout()
          crashlytics().recordError(error)
        })
    } else {
      setUser({})
    }
  }, [isAuthenticated, signout])

  const verifySubscription = useCallback(async () => {
    if (!subscription) return
    try {
      const params: Record<string, string | undefined> = { id: user.id }
      const status = await subscription.checkAccess(params)
      setSubscriptor(status)
    } catch (error) {
      if (error instanceof Error) crashlytics().recordError(error)
    }
  }, [user.id])

  useEffect(() => {
    if (typeof isAuthenticated === 'undefined') return
    if (isAuthenticated && !user.id) return
    if (isAuthenticated && user.id) {
      verifySubscription()
    } else {
      setSubscriptor(false)
    }
  }, [isAuthenticated, user.id, verifySubscription])

  useEffect(() => {
    if (typeof isAuthenticated === 'undefined') return
    if (isAuthenticated && !user.id) return
    const id = user.id || null
    analytics().setUserId(id)
    crashlytics().setUserId(id || '')
  }, [isAuthenticated, user.id])

  useEffect(() => {
    if (typeof subscriptor === 'undefined') return
    const type = subscriptor ? 'premium' : 'free'
    analytics().setUserProperty('account_type', type)
    sendPostMessageToWebview('Home', {
      type: 'subscription.UPDATE_STATUS',
      payload: { status: subscriptor },
    })
  }, [subscriptor])

  useEffect(() => {
    const callback = async (params: { url: string }) => {
      try {
        const config = auth.getOAuthConfig?.()
        if (!config) return
        if (!params.url.startsWith(config.redirect_url)) return
        const token = await tokenExchange(config, params.url)
        await auth.setToken(token)
        signin('oauth')
      } catch (_) {}
    }
    const listener = Linking.addEventListener('url', callback)
    return () => {
      listener.remove()
    }
  }, [signin])

  return (
    <AuthContext.Provider
      value={{
        isAuthenticated,
        isSubscribed: subscriptor,
        signin,
        signout,
        updateUserProfile,
        onAuthenticationChange,
        user,
        verifySubscription,
      }}>
      {children}
    </AuthContext.Provider>
  )
}

export const useAuth = (): Context => useContext(AuthContext)
