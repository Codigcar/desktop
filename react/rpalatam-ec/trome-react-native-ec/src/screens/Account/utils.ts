import { App } from '../../utils/config'
import * as Schema from '../../utils/validation'

const name = Schema.string()
  .matches(/\d+/i, 'Este campo no puede contener números')
  .matches(/\D{51,}/i, 'Máximo 50 caracteres')

export const ProfileSchema = Schema.object({
  first_name: App.select({
    elcomercio: name.clone().min(2).required(),
    default: name.clone().required(),
  }),
  last_name: name.clone().min(2).required(),
  second_last_name: name.clone().min(2),
  email: Schema.string().required().email(),
  mobile_phone: App.select({
    elcomercio: Schema.string()
      .matches(/\D+/i, 'Se aceptan solo números')
      .min(6, 'Mínimo 6 dígitos')
      .max(12, 'Máximo 12 dígitos'),
    default: Schema.string()
      .min(9, 'Mínimo 9 dígitos')
      .max(13, 'Máximo 13 dígitos'),
  }),
  user_metadata: {
    documentNumber: App.select({
      elcomercio: Schema.string().required(),
      default: Schema.string(),
    })
      .matches(/\s+/i, 'No se permiten espacios en blanco')
      .when('user_metadata.documentType', (value, schema) => {
        if (value === undefined) return schema
        if (value === 'DNI') {
          return schema
            .required()
            .matches(/\D+/i, 'Se aceptan solo números')
            .min(8, 'Se aceptan 8 dígitos')
            .max(8, 'Se aceptan 8 dígitos')
        }
        return schema
          .required()
          .min(9, 'Mínimo 9 dígitos')
          .max(20, 'Máximo 20 dígitos')
      }),
    documentType: Schema.string(),
  },
})
const gendersBase = [
  { label: 'Hombre', value: 'MALE' },
  { label: 'Mujer', value: 'FEMALE' },
]
export const OPTIONS = {
  country: [{ label: 'Perú', value: '260000' }],
  civilStatus: [
    { label: 'Soltero (a)', value: 'SO' },
    { label: 'Casado (a)', value: 'CA' },
    { label: 'Divorciado (a)', value: 'DI' },
    { label: 'Viudo (a)', value: 'VI' },
  ],
  document: [
    { label: 'DNI', value: 'DNI' },
    { label: 'CEX', value: 'CEX' },
    { label: 'CDI', value: 'CDI' },
  ],
  gender: App.select({
    elcomercio: [
      ...gendersBase,
      { label: 'Prefiero no decirlo', value: 'UNKNOWN' },
      { label: 'Otro', value: 'OTHER' },
    ],
    default: gendersBase,
  }),
}
