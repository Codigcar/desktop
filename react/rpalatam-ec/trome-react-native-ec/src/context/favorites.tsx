import AsyncStorage from '@react-native-async-storage/async-storage'
import analytics from '@react-native-firebase/analytics'
import React, {
  createContext,
  useCallback,
  useContext,
  useEffect,
  useMemo,
  useRef,
  useState,
} from 'react'

import { useAuth } from './auth'
import { Story } from '../entities/Story'
import Content, { queryInclude } from '../services/arc/Content'
import { getContentFormated } from '../services/arc/utils'
import auth from '../services/auth'
import { FavoritePreferences } from '../services/preferences'
import { App } from '../utils/config'
import { STORE_FAVORITES } from '../utils/constants'
import { HomeWebviewRef } from '../utils/refs'

interface Context {
  favorites: Story['id'][]
  fetchFavoritesStories: (size?: number) => Promise<Story[]>
  getFavorites: () => { ids: Story['id'][]; stories: Story[] }
  sendFavoritesToWebview: () => { ids: () => void; stories: () => void }
  setFavorites: () => {
    ids: (ids: Story['id'][]) => Promise<void>
    stories: (stories: Story[]) => void
  }
  stories: Story[]
  toggleFavorite: (story: Story['id']) => Promise<void>
}

export const FavoritesContext = createContext<Context>({} as Context)

export const FavoritesProvider: React.FC = ({ children }) => {
  const { isSubscribed, user } = useAuth()
  const [ids, setIds] = useState<Story['id'][]>([])
  const [listOfStories, setListOfStories] = useState<Story[]>([])
  const refIds = useRef<Story['id'][]>([])
  const refStories = useRef<Story[]>([])
  const Service = useMemo(() => {
    return Content.ids({
      params: { sourceInclude: queryInclude('cards').join() },
    })
  }, [])

  const sendFavoritesToWebview = useCallback(() => {
    function inject<T>(method: string, value: T) {
      HomeWebviewRef.current?.injectJavaScript(
        `(function() {
          window.NATIVE_CONNECTION && window.NATIVE_CONNECTION.favorites['${method}'](${JSON.stringify(
          value,
        )})
        })()`,
      )
    }
    return {
      ids() {
        inject<Story['id'][]>('setIds', refIds.current)
      },
      stories() {
        inject<Story[]>('setStories', refStories.current)
      },
    }
  }, [])

  useEffect(() => {
    sendFavoritesToWebview().ids()
  }, [ids, sendFavoritesToWebview])

  useEffect(() => {
    sendFavoritesToWebview().stories()
  }, [listOfStories, sendFavoritesToWebview])

  useEffect(() => {
    const nextStories = refStories.current.filter((prevStory) =>
      ids.includes(prevStory.id),
    )
    refStories.current = nextStories
  }, [ids])

  const getFavorites = useCallback(() => {
    return { ids: refIds.current, stories: refStories.current }
  }, [])

  const setFavoritesIds = useCallback((storiesIds: Story['id'][]) => {
    refIds.current = storiesIds
    setIds(storiesIds)
  }, [])

  const setFavoritesStories = useCallback((stories: Story[]) => {
    refStories.current = stories
    setListOfStories(stories)
  }, [])

  const saveFavoritesIds = useCallback(
    async (storiesIds: string[]) => {
      if (isSubscribed && App.key !== 'elcomercio') {
        FavoritePreferences.abort()
        const token = await auth.getToken()
        FavoritePreferences.post(storiesIds, {
          accessToken: token?.access_token,
          email: user.email,
          id: user.id,
        }).catch(console.log)
      } else {
        await AsyncStorage.setItem(STORE_FAVORITES, JSON.stringify(storiesIds))
      }
      setFavoritesIds(storiesIds)
    },
    [isSubscribed, setFavoritesIds, user],
  )

  const setFavorites = useCallback(() => {
    return {
      ids(storiesIds: Story['id'][]) {
        return saveFavoritesIds(storiesIds)
      },
      stories(stories: Story[]) {
        return setFavoritesStories(stories)
      },
    }
  }, [saveFavoritesIds, setFavoritesStories])

  const toggleFavorite = useCallback(
    async (storyId: Story['id']) => {
      const isFavorite = refIds.current.includes(storyId)

      if (!isFavorite) {
        await analytics().logEvent('add_to_read_later', {
          id: storyId,
        })
      }

      const nextFavorites = isFavorite
        ? refIds.current.filter((id) => id !== storyId)
        : [...refIds.current, storyId]
      await saveFavoritesIds(nextFavorites)
    },
    [saveFavoritesIds],
  )
  const fetchFavoritesStories = useCallback(
    async (size) => {
      Service.abort()
      let currentIds = Array.from(refIds.current).reverse()
      let idsFetched = refStories.current.map((story) => story.id)

      if (App.key === 'elcomercio') {
        currentIds = currentIds.splice(0, size)
        idsFetched = idsFetched.splice(0, size)
      }

      const storiesForFetch = currentIds.filter(
        (id) => !idsFetched.includes(id),
      )
      if (storiesForFetch.length === 0) return refStories.current
      const response = await Service.get(storiesForFetch)
      const storiesFormated = getContentFormated(response)
      const newStories = [...refStories.current, ...storiesFormated]
      return currentIds.map((id) =>
        newStories.find((story) => story.id === id),
      ) as Story[]
    },
    [Service],
  )

  const readFavoritesFromStorage = useCallback(async () => {
    const item = await AsyncStorage.getItem(STORE_FAVORITES)
    if (item === null) return setFavoritesIds([])
    const favorites: Story['id'][] = JSON.parse(item)
    setFavoritesIds(favorites)
  }, [setFavoritesIds])

  useEffect(() => {
    if (isSubscribed && user.id && App.key !== 'elcomercio') {
      FavoritePreferences.get(user.id)
        .then((remoteIds) => {
          const listOfIds = remoteIds.filter((id) => !!id)
          setFavoritesIds(listOfIds)
        })
        .catch(console.log)
    } else {
      readFavoritesFromStorage()
    }
  }, [isSubscribed, readFavoritesFromStorage, setFavoritesIds, user.id])

  return (
    <FavoritesContext.Provider
      value={{
        favorites: ids,
        getFavorites,
        fetchFavoritesStories,
        sendFavoritesToWebview,
        setFavorites,
        stories: listOfStories,
        toggleFavorite,
      }}>
      {children}
    </FavoritesContext.Provider>
  )
}

export const useFavorites = (): Context => useContext(FavoritesContext)
