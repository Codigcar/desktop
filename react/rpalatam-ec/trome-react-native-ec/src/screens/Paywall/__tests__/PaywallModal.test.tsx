import { useNavigation, useRoute } from '@react-navigation/native'
import { act, fireEvent, render } from '@testing-utils/library'
import React from 'react'

import { useAuth } from '../../../context/auth'
import { openInBrowser } from '../../../utils/inappbrowser'
import PaywallModalScreen from '../PaywallModal'

jest.mock('@react-navigation/native')
const mockUseNavigation = useNavigation as jest.Mock
mockUseNavigation.mockReturnValue({})
const mockUseRoute = useRoute as jest.Mock
mockUseRoute.mockReturnValue({ params: { title: 'Title' } })

jest.mock('../../../context/auth')
const mockUseAuth = useAuth as jest.Mock
mockUseAuth.mockReturnValue({})

jest.mock('../../../services/subscription', () => ({
  plan: jest.fn().mockResolvedValue({
    id: 'id',
    name: 'Plan Digital',
    interval: 'month',
    interval_count: 1,
    offer: { price: 5, interval: 'month', interval_count: 1 },
    price: 20,
  }),
}))

jest.mock('../../../utils/inappbrowser')

describe('PaywallModal', () => {
  it('should render plan', async () => {
    const { toJSON } = render(<PaywallModalScreen />)
    await act(async () => undefined)
    expect(toJSON()).toMatchSnapshot()
  })

  it('should open landing page', async () => {
    const { getByText } = render(<PaywallModalScreen />)
    await act(async () => undefined)
    fireEvent.press(getByText('VER PLANES'))
    expect(openInBrowser).toBeCalledTimes(1)
  })

  it('should navigate to login screen when user is not logged in', async () => {
    mockUseAuth.mockReturnValue({ user: {} })
    const navigate = jest.fn()
    mockUseNavigation.mockReturnValue({ navigate })
    const { getByText } = render(<PaywallModalScreen />)
    await act(async () => undefined)
    fireEvent.press(getByText('Restaurar'))
    expect(navigate).toBeCalledWith('Login')
  })

  it('should verify subscription', async () => {
    const verifySubscription = jest.fn().mockResolvedValue(undefined)
    mockUseAuth.mockReturnValue({ user: { id: 'id' }, verifySubscription })
    const { getByText, queryByText } = render(<PaywallModalScreen />)
    await act(async () => undefined)
    fireEvent.press(getByText('Restaurar'))
    expect(getByText('Validando...')).toBeDefined()
    expect(verifySubscription).toBeCalled()
    await act(async () => undefined)
    expect(queryByText('Validando...')).toBeNull()
  })

  it('should navigate to back when modal is closed', async () => {
    const goBack = jest.fn()
    mockUseNavigation.mockReturnValue({ goBack })
    const { getByA11yLabel } = render(<PaywallModalScreen />)
    await act(async () => undefined)
    fireEvent.press(getByA11yLabel('cerrar'))
    expect(goBack).toBeCalled()
  })
})
