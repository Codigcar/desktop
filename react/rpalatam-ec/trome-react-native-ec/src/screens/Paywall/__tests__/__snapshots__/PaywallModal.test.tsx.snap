// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaywallModal should render plan 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RCTScrollView
    bounces={false}
    style={
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
          "display": "flex",
          "flex": 1,
          "paddingTop": 16,
        },
      ]
    }
  >
    <View>
      <View
        style={
          Array [
            Object {
              "display": "flex",
              "paddingHorizontal": 20,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignSelf": "flex-end",
                "display": "flex",
                "marginBottom": 4,
              },
            ]
          }
        >
          <View
            accessible={true}
            focusable={true}
            hitSlop={
              Object {
                "bottom": 4,
                "left": 4,
                "right": 4,
                "top": 4,
              }
            }
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "display": "flex",
                  "height": 12,
                  "width": 12,
                },
              ]
            }
          >
            <SvgMock
              accessibilityLabel="cerrar"
              fill="#C1C1C1"
            />
          </View>
        </View>
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
                "display": "flex",
                "fontFamily": "NotoSans",
                "fontSize": 18,
                "fontWeight": "700",
                "lineHeight": 24,
                "textAlign": "center",
              },
              Object {
                "fontFamily": "NotoSans-Bold",
              },
            ]
          }
        >
          Title
        </Text>
        <View
          style={
            Array [
              Object {
                "display": "flex",
                "paddingVertical": 12,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#3C3C3C",
                  "display": "flex",
                  "fontFamily": "NotoSans",
                  "fontSize": 30,
                  "fontWeight": "300",
                  "lineHeight": 32,
                  "textAlign": "center",
                },
                Object {
                  "fontFamily": "NotoSans-Light",
                },
              ]
            }
          >
            Elige el plan que más te convenga
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "row",
                "justifyContent": "center",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#8E8E8E",
                  "display": "flex",
                  "fontFamily": "NotoSans",
                  "fontSize": 16,
                  "lineHeight": 24,
                  "textAlign": "center",
                },
                Object {
                  "fontFamily": "NotoSans-Regular",
                },
              ]
            }
          >
            ¿Ya tienes una suscripción?
             
          </Text>
          <View
            accessible={true}
            focusable={true}
            hitSlop={
              Object {
                "bottom": 4,
                "left": 4,
                "right": 4,
                "top": 4,
              }
            }
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "flexDirection": "row",
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#0089ff",
                    "display": "flex",
                    "fontFamily": "NotoSans",
                    "fontSize": 16,
                    "lineHeight": 24,
                    "textAlign": "center",
                  },
                  Object {
                    "fontFamily": "NotoSans-Regular",
                  },
                ]
              }
            >
              Restaurar
            </Text>
            <View
              style={
                Array [
                  Object {
                    "display": "flex",
                    "height": 8,
                    "marginLeft": 4,
                    "top": 1,
                    "width": 8,
                  },
                ]
              }
            >
              <SvgMock
                fill="#1F92E6"
              />
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#FFCB05",
              "display": "flex",
              "elevation": 3,
              "marginBottom": 16,
              "marginTop": 40,
              "paddingBottom": 32,
              "paddingHorizontal": 24,
              "paddingTop": 16,
              "shadowColor": "#050505",
              "shadowOffset": Object {
                "height": 3,
                "width": 0,
              },
              "shadowOpacity": 0.32,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "display": "flex",
                "fontFamily": "NotoSans",
                "fontSize": 20,
                "fontWeight": "900",
                "lineHeight": 24,
                "textAlign": "center",
                "textTransform": "uppercase",
              },
              Object {
                "fontFamily": "NotoSans-Black",
              },
            ]
          }
        >
          Plan Digital Mensual
        </Text>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "row",
                "justifyContent": "center",
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "flexDirection": "row",
                  "marginRight": 12,
                },
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "display": "flex",
                    "top": 12,
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#050505",
                      "display": "flex",
                      "fontFamily": "NotoSans",
                      "fontSize": 36,
                      "fontWeight": "900",
                      "lineHeight": 40,
                    },
                    Object {
                      "fontFamily": "NotoSans-Black",
                    },
                  ]
                }
              >
                s/
              </Text>
            </View>
            <Text
              style={
                Array [
                  Object {
                    "color": "#050505",
                    "display": "flex",
                    "fontFamily": "NotoSans",
                    "fontSize": 60,
                    "fontWeight": "900",
                  },
                  Object {
                    "fontFamily": "NotoSans-Black",
                  },
                ]
              }
            >
              5
            </Text>
          </View>
          <Text
            style={
              Array [
                Object {
                  "color": "#050505",
                  "display": "flex",
                  "fontFamily": "NotoSans",
                  "fontSize": 20,
                  "fontWeight": "500",
                  "lineHeight": 24,
                },
                Object {
                  "fontFamily": "NotoSans-Medium",
                },
              ]
            }
          >
            El primer
            

            mes
          </Text>
        </View>
        <Text
          style={
            Array [
              Object {
                "color": "#050505",
                "display": "flex",
                "fontFamily": "NotoSans",
                "fontSize": 20,
                "lineHeight": 24,
                "textAlign": "center",
              },
              Object {
                "fontFamily": "NotoSans-Regular",
              },
            ]
          }
        >
          Luego S/
          20
           al mes
        </Text>
        <View
          style={
            Array [
              Object {
                "display": "flex",
                "marginTop": 20,
              },
            ]
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#050505",
                  "borderRadius": 6,
                  "display": "flex",
                  "elevation": 3,
                  "flexDirection": "row",
                  "flexWrap": "nowrap",
                  "height": 48,
                  "justifyContent": "center",
                  "opacity": 1,
                  "shadowColor": "#050505",
                  "shadowOffset": Object {
                    "height": 3,
                    "width": 0,
                  },
                  "shadowOpacity": 0.32,
                  "shadowRadius": 6,
                  "width": "100%",
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "display": "flex",
                    "fontFamily": "NotoSans",
                    "fontSize": 18,
                    "fontWeight": "700",
                    "lineHeight": 28,
                  },
                  Object {
                    "fontFamily": "NotoSans-Bold",
                  },
                ]
              }
            >
              VER PLANES
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "display": "flex",
              "marginBottom": 32,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#3C3C3C",
                "display": "flex",
                "fontFamily": "NotoSans",
                "fontSize": 20,
                "fontWeight": "700",
                "lineHeight": 32,
                "textAlign": "center",
              },
              Object {
                "fontFamily": "NotoSans-Bold",
              },
            ]
          }
        >
          Contenidos exclusivos
          

          +
          

          Navegación ilimitada
          

          +
          

          Club de Suscriptores
          

          +
          

          Sin anuncios en la app
        </Text>
      </View>
    </View>
  </RCTScrollView>
</RNCSafeAreaProvider>
`;
